### PR TITLE
Fix install-docker role

### PR DESCRIPTION
**Pull request by kiwik**
1. systemctl is not available on Ubuntu 14.04, so using
   "service docker restart" as a workaround.
2. change option "--r" position in spark-inegration-test-minikube-k8s
   job to avoid "--r" is bypassed to mvn.
3. change the importing order of roles in docker-machine job to make sure the
  docker installation won't clean the iptables rules added by devstack

Related-Bug: theopenlab/openlab#308
Related-Bug: theopenlab/openlab#310

Co-authored-by: liu-sheng <liusheng2048@gmail.com>

### CODE REVIEW

File: roles/install-docker/tasks/main.yml [hunk 1]
```diff
@@ -27,8 +27,14 @@
       iptables -P FORWARD ACCEPT
       iptables -P OUTPUT ACCEPT
 
-      # Restart docker.
-      systemctl daemon-reload
-      systemctl restart docker
+      # NOTE: systemctl is not available in Ubuntu 14.04
```
Comment (liu-sheng) [inline roles/install-docker/tasks/main.yml]: maybe just use `service docker restart` ?

Comment (kiwik) [reply roles/install-docker/tasks/main.yml]: OK, good point.

**Comment by liu-sheng** [comment]
sorry, according to my testing, these jobs still need docker installed execution node.

File: playbooks/docker-machine-functional-devstack/run.yaml [hunk 2]
```diff
@@ -1,10 +1,10 @@
 - hosts: all
   become: yes
   roles:
+    - config-docker-machine
     - clone-devstack-gate-to-workspace
     - create-devstack-local-conf
     - install-devstack
-    - config-docker-machine
```
Comment (bzhaoopenstack) [inline playbooks/docker-machine-functional-devstack/run.yaml]: Just want to make sure, does it pass in allinone? Others looks good.

Comment (liu-sheng) [reply playbooks/docker-machine-functional-devstack/run.yaml]: yes, you can see: http://80.158.17.129/builds?uuid=f62c7423653d41a2a9cff9777cb4ce3a

**Comment by kiwik** [comment]
LGTM, but I'am PR owner, so can't approve

**Pull request merged**